querybook/server/app/auth/__init__.py

```diff
@@ -1,14 +1,32 @@
+import time
+
 from flask import request
 from flask_login import logout_user, current_user
 
 from const.path import BUILD_PATH
 from const.datasources import DS_PATH
 from env import QuerybookSettings
 from lib.utils.import_helper import import_module_with_default
+from lib.stats_logger import stats_logger, ACTIVE_USER_COUNTER
 
 
 auth = None
 login_config = None
+active_users = {}
+
+ACTIVE_USER_SESSION_TIMEOUT = 30 * 60  # 30 minutes in seconds
+
+
+def update_active_users():
+    if current_user.id not in active_users:
+        stats_logger.incr(ACTIVE_USER_COUNTER)
+
+    active_users[current_user.id] = time.time()
+
+    for user_id, last_activity_time in list(active_users.items()):
+        if time.time() - last_activity_time > ACTIVE_USER_SESSION_TIMEOUT:
```

Comment:
maybe we don't need to log the num of active users? 3 reasons:
1. it is quite annoying to handle with multiple servers
2. the stat is not as useful as number of requests which we also have
3. the num of user can also be extracted from impression log
4. this call is not real time, which pads a variable processing time to the request